feat: Added Ollama in the project. Requirements updated and freezed.

CasinoRoyal666 opened the pull request:
docker-compose.yml file updated with ollama service translate_question command that translates and gives PEGI rating to quiz added in management/commands/ directory Added translate_all.bat into root directory

CasinoRoyal666 committed:
Initial Commit: Django

CasinoRoyal666 committed:
Added Item model, serializer, CRUD views on Item model
Added urls.py in api/ project folder
Changed settings.py
Added couple of API tests based on APITestCase class

CasinoRoyal666 committed:
Merge pull request #1 from CasinoRoyal666/feature/backend_base_init

Backend initialization

CasinoRoyal666 committed:
feat: Added prod/dev modes
Added settings/ folder in config/. Settings folder have settings for dev and prod modes, also base broject settings
.env.example updated
now project uses PostgreSQL

CasinoRoyal666 committed:
Created Next.js application

CasinoRoyal666 committed:
Merge pull request #2 from CasinoRoyal666/feature/dev_prod_modes

feat: Added prod/dev modes

CasinoRoyal666 committed:
feat: Added basic interface for ItemList, ItemDetails
Now user cn perform CRUD operations on items
Fix development.py file

CasinoRoyal666 committed:
feat: Visual improvements of all components, ItemList now is Server Component

CasinoRoyal666 committed:
Merge pull request #3 from CasinoRoyal666/feature/frontend_app

Feature/frontend app

CasinoRoyal666 committed:
feat: Added nginx configs, added Dockerfile for frontend/ and backend/ folders
added docker-compose.yml and Makefile in root directory
Deleted page.tsx in app/items
Updated frontend components
Added .env.examples

CasinoRoyal666 committed:
Merge pull request #4 from CasinoRoyal666/feature/nginx_docker

feat: Added nginx configs, added Dockerfile for frontend/ and backend…

CasinoRoyal666 committed:
feat: better broject structure
Added enviroment profiles into docker-compose.yml
Changed .env.dev and .env.prod (also .env.examples updated)
Added black, isort, flake8 linters
Requirements file updated
Added .eslintrc.js, .prettierrc files into frontend/ folders (project.json updated)
Changed PostgreSQL ver 16 -> 17
Changed backend/Dockerfile. Logic moved to new entrypoint.sh file
Changed python version 3.11-slim -> 3.12-slim(stable)

CasinoRoyal666 committed:
feat/fix: multiple changes and corrections
Now dev/prod modes are 100% working
Created .bat files for convenient starting and stopping of containers
Added Dockerfile.prod and Dockerfile.dev in frintend/ folder
Changed docker-compose.yml, entrypoint.sh, nginx files, backend settings files

CasinoRoyal666 committed:
Merge pull request #5 from CasinoRoyal666/feature/nginx_docker

feat: better broject structure

CasinoRoyal666 committed:
feat: Added 2 new models - Category and Question for next parsing stage
Added models to admin.py
Added new structure to the backend/ - management/commands/ folder with parsing_commands file
Added selenuim packcage, requirements freezed
Added selenium to docker-compose

CasinoRoyal666 committed:
test: Testing a simple parsing. parsing_command file changed

CasinoRoyal666 committed:
feat: Parser now parses! The model has been modified to meet parsing requirements.

CasinoRoyal666 committed:
fix: Fixed parsing command: now command parses trivia site correcrly, assigning a category and the correct answer to each quiz
Added Category and Question models to admin.py for checking data

CasinoRoyal666 committed:
Merge pull request #6 from CasinoRoyal666/feature/task2_trivia_parse

feat: Task 2- Parsing

CasinoRoyal666 committed:
feat: Added Ollama in the project. Requirements updated and freezed.
docker-compose.yml file updated with ollama service
translate_question command that translates and gives PEGI rating to quiz added in management/commands/ directory
Added translate_all.bat into root directory

panfillich commented on backend/api/management/commands/parsing_command.py:
```diff
@@ -68,11 +68,11 @@ def handle(self, *args, **options):
 
                             if category_name:
                                 category, _ = Category.objects.get_or_create(name=category_name)
-                                self.stdout.write(f'  📂 Category: "{category_name}"')
```
Why parsing command already exist in main?

CasinoRoyal666 replied:
There was a bug in the code: the check for the existence of the question in the database was performed after the category and answer had been parsed. The problem has been fixed. Is that what you meant?

CasinoRoyal666 replied:
👍

panfillich commented on backend/api/management/commands/parsing_command.py:
```diff
@@ -82,53 +82,51 @@ def handle(self, *args, **options):
                             if correct_answer:
                                 self.stdout.write(f'  ✓ Answer: "{correct_answer[:60]}..."')
                             else:
-                                self.stdout.write(self.style.WARNING(f'  ⚠️  Empty answer text'))
```
I asked to add linters and pre-commit hooks. Why are you ignore it

panfillich commented on backend/requrements.txt:
Why you remove this file! I want to see changes

CasinoRoyal666 replied:
You can see all changes in backend/requirements.txt file. That was a misspelled version of the right file

panfillich commented on translate_all.bat:
```diff
@@ -0,0 +1,42 @@
+@echo off
```
Why do we need this? I dont want to see anything related to window or any OS related things

CasinoRoyal666 replied:
Added console commands to run translate_command

CasinoRoyal666 replied:
Deleted all .bat files. Check README for startapp info

CasinoRoyal666 committed:
fix: Adde pre-commit to the project, fixed pointed by pre-commit problem
Removed translate_all.bat
Updated README.md and requierements.txt
Fixed a bug in parsing_command.py

CasinoRoyal666 committed:
fix: Deleted OS related files run-dev.bat, run-prod.bat, stop.bat. Fixed a requrements and development settings files
